lib/connectors/base/connector.rb

```diff
@@ -33,11 +34,24 @@ def initialize(local_configuration: {}, remote_configuration: {})
 
       def yield_documents; end
 
-      def source_status(params = {})
-        health_check(params)
-        { :status => 'OK', :statusCode => 200, :message => "Connected to #{self.class.display_name}" }
+      def do_health_check(_params)
+        raise 'Not implemented for this connector'
+      end
+
+      def do_health_check!(params)
```

Comment:
I'm thinking we should make `do_health_check!` and `is_healthy?` non-overridable. But it seems there's no `final` keyword in Ruby as in Java.

Another point I would like to discuss, our `connectors-ruby` is mixed of connector service (the framework) and the connectors (the logic of each connector, in dir `lib/connectors`). 

The `Connectors::Base::Connector` is the only ambiguous file. It's in dir `lib/connectors`, but we certainly don't want anyone who wants to customize to modify this file.

Just my random thoughts, and I don't have any better suggestions.

Reply:
> I'm thinking we should make do_health_check! and is_healthy? non-overridable. But it seems there's no final keyword in Ruby as in Java.

I don't think it's possible indeed, the only way to make them non-overridable is not exposing them as a part of a base class, therefore splitting the abstraction into two, and I honestly think it's the best way to move forward, we can think how to make it nice in a different PR.

> Another point I would like to discuss, our connectors-ruby is mixed of connector service (the framework) and the connectors (the logic of each connector, in dir lib/connectors).

The Connectors::Base::Connector is the only ambiguous file. It's in dir lib/connectors, but we certainly don't want anyone who wants to customize to modify this file.

+1 to that, very very important points. I'd +1 moving them away and leaving only actual connectors in the `connectors` directory.